### PR TITLE
library page: playlists

### LINKED ISSUE #16: library page: playlists
We want the ability for the user to be able to see a page of the playlists they have saved. This makes it easier for them to collect the songs and albums that they enjoy, and to give them easy access to what they have saved. Giving the user this kind of fast access, increases user engagement, and keeps them happy and subscribed.

- [ ] large component showing how many liked songs, a preview of artists/tracks, that when clicked brings to giant playlists of liked songs
- [ ] a collections of playlists displayed as cards, when clicked, brings to playlists interface

### PR DESCRIPTION
**Pull request by chelsey-g**
## Issue ticket number and link

Fixes #16.

## Checklist before requesting a review
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics?
- [ ] Will this be part of a product update? If yes, please write one phrase about this update.

### CODE REVIEW

**Comment by chelsey-g** [comment]
1. I added CSS grid to Playlist view.
2. I added CSS flex to Playlist view header.
3. Fixed minor CSS issues.

File: src/components/CategoriesView.jsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import "./CategoriesView.css";
+import { Link } from "react-router-dom";
```
Comment (phishy) [inline src/components/CategoriesView.jsx:2]: This import is unused. No need for it to be here. VSCode should show it in a lightened color to indicat that it is unused.

File: src/components/CategoriesView.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        />
+        <div className="category-title-subheader-2">Classics</div>
+        <MediumCardCollection
+          playlists={CategoryPlaylists}
+          className="playlist-card"
+        />
+      </div>
+    </div>
+  );
+}
```
Comment (phishy) [inline src/components/CategoriesView.jsx:29]: Adding PropTypes to this component will help you and the next developer know what props it accepts.

File: src/components/CategoriesView.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import "./CategoriesView.css";
+import { Link } from "react-router-dom";
+import MediumCardCollection from "./MediumCardCollection";
+
+export default function CategoriesView(props) {
+  let { CategoryType, CategoryPlaylists } = props;
```
Comment (phishy) [inline src/components/CategoriesView.jsx:6]: In this project, and in most, usually lowercase first letters are used for hook variable names.

File: src/components/LargeCard.jsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+      <div className="largecard-playlistinfo">Liked Songs</div>
+      <div className="largecard-totaltracks">
+        {props.likedSongs.length} liked songs
+      </div>
+    </div>
+  );
+}
+
+LargeCard.propTypes = {
+  playlist: PropTypes.object.isRequired,
```
Comment (phishy) [inline src/components/LargeCard.jsx:31]: It looks like this component accepts a `likedSongs` prop, which you can document here in the proptypes. It also looks like you are not using the `playlist` prop. Do we still need it?

File: src/components/LeftArrowButton.jsx [hunk 5]
```diff
@@ -16,7 +16,7 @@ export default function LeftArrowButton(props) {
       className={`left-arrow-button ${LeftArrowClass}`}
```
Comment (phishy) [inline src/components/LeftArrowButton.jsx:16]: `LeftArrowClass` variable above, should be `leftArrowClass` to fit with the rest of the project's style.

File: src/pages/CategoriesPage.jsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { getCategoryTitle } from "../util/spotify";
+import { getCategoryPlaylists } from "../util/spotify";
+import CategoriesView from "../components/CategoriesView";
+
+export default function CategoriesPage(props) {
+  const { id } = useParams();
+  const [CategoryType, setCategoryType] = useState();
```
Comment (phishy) [inline src/pages/CategoriesPage.jsx:9]: `CategoryType` and `CategoryPlaylists` should start with lowercase letters as per project style

File: src/pages/CategoriesPage.jsx [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }, [id]);
+
+  useEffect(() => {
+    getCategoryPlaylists(id)
+      .then((data) => {
+        console.log("data loaded");
```
Comment (phishy) [inline src/pages/CategoriesPage.jsx:26]: Shouldnt leave around console.log debugging lines.

File: src/pages/SearchPage.jsx [hunk 8]
```diff
@@ -1,20 +1,27 @@
 import React, { useEffect, useState } from "react";
 import { getSearchItems } from "../util/spotify";
-
+import { getBrowseCategories } from "../util/spotify";
 import MainBody from "../components/MainBody";
```
Comment (phishy) [inline src/pages/SearchPage.jsx:4]: This import is unused.

**Review by phishy — COMMENTED** [review]
This PR looks great! I left a bunch of comments about proptypes, unused imports, and variable naming style fo the project.

Also I got errors trying to run this even after I did `npm i`.. turns out I had to install `@material-ui/core` and `@material-ui/icons`. These should be added to your package.json and committed.

Also, I get an error when trying to click on albums now.

**Review by chelsey-g — COMMENTED** [review]
I have gone ahead and fixed all the minor errors based off your comments. I know know of my simple mistakes that can easily be avoided.

**Pull request merged**